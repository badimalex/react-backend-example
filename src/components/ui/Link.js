import { Link } from 'react-router';

export default Link;
