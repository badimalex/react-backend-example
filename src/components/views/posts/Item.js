import React from 'react';
import PropTypes from 'prop-types';

import ItemFull from 'components/widgets/blog/ItemFull';

import Helmet from 'react-helmet';

import { Grid } from 'semantic-ui-react';

const Post = ({ item, actions }) => (
  <Grid>
  <Grid.Row columns={1}>
      <Grid.Column>
        {item && <ItemFull {...item}  addCommentHandler={actions.addComment} />}
        {item && <Helmet title={item.title} /> }
        </Grid.Column>
    </Grid.Row>
  </Grid>
);

Post.propTypes = {
  item: PropTypes.object
};

export default Post;
