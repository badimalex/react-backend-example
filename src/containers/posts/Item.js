import { connect } from 'react-redux';
import Post from 'components/views/posts/Item';
import { bindActionCreators } from 'redux';
import * as PostActions from 'actions/Post';

const stateToProps = (state) => ({
  item: state.post.entry,
  isFetching: state.post.isFetching,
  error: state.post.error
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(PostActions, dispatch)
});

export default connect(
  stateToProps,
  mapDispatchToProps
)(Post);
