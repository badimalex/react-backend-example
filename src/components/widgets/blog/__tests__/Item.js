import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Link from 'components/ui/Link';
import Item from '../ItemShort';

describe('Item', () => {
  it('renders fine', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Item />, div);
  });
  describe('render', () => {
    it('should render the title', () => {
      const itemProps = { title: 'hi', postUrl: '/posts/5' };
      const item = shallow(<Item {...itemProps} />);
      const header = <Link to="/posts/5">hi</Link>;
      expect(item.contains(header)).toEqual(true);
    });

    it('should render usual item', () => {
      const itemProps = {
        title: 'hello',
        postUrl: '/posts/5',
        image: {
          src: '/images/google.png'
        }
      };
      const item = shallow(<Item {...itemProps} />);
      expect(item).toMatchSnapshot();
    });
  });
});
