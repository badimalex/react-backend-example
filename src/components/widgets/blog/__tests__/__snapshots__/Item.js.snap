// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item render should render usual item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Item>
    <ItemImage
        src="/images/google.png"
    />
    <ItemContent>
        <ItemHeader>
            <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/posts/5"
            >
                hello
            </Link>
             
            <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to={undefined}
            >
                <Icon
                    as="i"
                    name="edit"
                />
            </Link>
        </ItemHeader>
        <ItemDescription>
             
             
        </ItemDescription>
        <ItemMeta>
            by undefined,
                      on 05/10/2017,
                      updated 05/10/2017
        </ItemMeta>
        <Like
            likeHandler={undefined}
            likes={undefined}
        />
    </ItemContent>
</Item>,
  "nodes": Array [
    <Item>
      <ItemImage
            src="/images/google.png"
      />
      <ItemContent>
            <ItemHeader>
                  <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/posts/5"
                  >
                        hello
                  </Link>
                   
                  <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to={undefined}
                  >
                        <Icon
                              as="i"
                              name="edit"
                        />
                  </Link>
            </ItemHeader>
            <ItemDescription>
                   
                   
            </ItemDescription>
            <ItemMeta>
                  by undefined,
                            on 05/10/2017,
                            updated 05/10/2017
            </ItemMeta>
            <Like
                  likeHandler={undefined}
                  likes={undefined}
            />
      </ItemContent>
</Item>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        image={
                Object {
                        "src": "/images/google.png",
                      }
        }
        postUrl="/posts/5"
        title="hello"
/>,
      "_debugID": 38,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "image": Object {
            "src": "/images/google.png",
          },
          "postUrl": "/posts/5",
          "title": "hello",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Item>
          <ItemImage
                    src="/images/google.png"
          />
          <ItemContent>
                    <ItemHeader>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        hello
                              </Link>
                               
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to={undefined}
                              >
                                        <Icon
                                                  as="i"
                                                  name="edit"
                                        />
                              </Link>
                    </ItemHeader>
                    <ItemDescription>
                               
                               
                    </ItemDescription>
                    <ItemMeta>
                              by undefined,
                                        on 05/10/2017,
                                        updated 05/10/2017
                    </ItemMeta>
                    <Like
                              likeHandler={undefined}
                              likes={undefined}
                    />
          </ItemContent>
</Item>,
        "_debugID": 39,
        "_renderedOutput": <Item>
          <ItemImage
                    src="/images/google.png"
          />
          <ItemContent>
                    <ItemHeader>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        hello
                              </Link>
                               
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to={undefined}
                              >
                                        <Icon
                                                  as="i"
                                                  name="edit"
                                        />
                              </Link>
                    </ItemHeader>
                    <ItemDescription>
                               
                               
                    </ItemDescription>
                    <ItemMeta>
                              by undefined,
                                        on 05/10/2017,
                                        updated 05/10/2017
                    </ItemMeta>
                    <Like
                              likeHandler={undefined}
                              likes={undefined}
                    />
          </ItemContent>
</Item>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    image={
        Object {
            "src": "/images/google.png",
          }
    }
    postUrl="/posts/5"
    title="hello"
/>,
}
`;
